applications/trilinos_application/external_includes/mpi_amgcl_ns_solver.h

```diff
@@ -0,0 +1,391 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:		 BSD License
+//					 Kratos default license: kratos/license.txt
+//
+//  Main authors:    Denis Demidov
+//                   Riccardo Rossi
+//
+
+
+#if !defined(KRATOS_AMGCL_NS_MPI_SOLVER_H_INCLUDED )
+#define  KRATOS_AMGCL_NS_MPI_SOLVER_H_INCLUDED
+
+
+#ifndef AMGCL_PARAM_UNKNOWN
+#  define AMGCL_PARAM_UNKNOWN(name)                                            \
+      std::cerr << "AMGCL WARNING: unknown parameter " << name << std::endl
+#endif
+
+
+// External includes
+#include "boost/smart_ptr.hpp"
+
+// Project includes
+#include "includes/define.h"
+#include "includes/kratos_parameters.h"
+#include "linear_solvers/linear_solver.h"
+#include "external_includes/amgcl_solver.h"
+
+//aztec solver includes
+#include "AztecOO.h"
+#include "Epetra_LinearProblem.h"
+//#include "Teuchos_ParameterList.hpp"
+
+
+#include <amgcl/amg.hpp>
+#include <amgcl/adapter/epetra.hpp>
+#include <amgcl/profiler.hpp>
+
+#include <amgcl/mpi/make_solver.hpp>
+#include <amgcl/mpi/schur_pressure_correction.hpp>
+#include <amgcl/mpi/block_preconditioner.hpp>
+#include <amgcl/mpi/subdomain_deflation.hpp>
+#include <amgcl/mpi/direct_solver.hpp>
+#include <boost/property_tree/json_parser.hpp> //needed to print AMGCL internal settings
+
+
+
+namespace Kratos
+{
+
+
+
+template< class TSparseSpaceType, class TDenseSpaceType,
+          class TReordererType = Reorderer<TSparseSpaceType, TDenseSpaceType> >
+class AmgclMPI_NS_Solver : public LinearSolver< TSparseSpaceType,
```

Comment:
changed name to 
AmgclMPISchurComplementSolver 

which is more accurate (it is not just for navier stokes, it also applies to stokes or to any mixed u-p d-p problem)